src/features/diagnosticsProvider.ts

```diff
@@ -162,17 +162,17 @@ class DiagnosticsProvider extends AbstractSupport {
     }
 
     private _onDocumentRemove(document: vscode.TextDocument) {
-        let key = document.uri.toString();
+        let key = document.uri;
         let didChange = false;
-        if (this._diagnostics[key]) {
+        if (this._diagnostics.get(key)) {
             didChange = true;
-            this._diagnostics[key].dispose();
-            delete this._diagnostics[key];
+            this._diagnostics.delete(key);
         }
-        if (this._documentValidations[key]) {
+        
+        if (this._documentValidations[key.toString()]) {
```

Comment:
I think we did `key.toString()` enough times to justify introducing a local :)
BTW, why is `key` directly used as a ... key in `_diagnostics` but we toString it to use it as a key in `_documentValidations`?